web3.js/src/connection.ts

```diff
@@ -2580,11 +2581,27 @@ export class Connection {
     publicKeys: PublicKey[],
     commitment?: Commitment,
   ): Promise<(AccountInfo<Buffer> | null)[]> {
-    const res = await this.getMultipleAccountsInfoAndContext(
-      publicKeys,
-      commitment,
-    );
-    return res.value;
+    const keys = publicKeys.map(key => key.toBase58());
+    const chunks = sliceIntoChunks(keys, 100);
+    let accounts : (AccountInfo<Buffer> | null)[] = [];
+
+    for (let i = 0; i < chunks.length; i++) {
+      const args = this._buildArgs([chunks[i]], commitment, 'base64');
+      const unsafeRes = await this._rpcRequest('getMultipleAccounts', args);
```

Comment:
The thing about calling `await` in a loop is that each successive loop has to wait on the last one to finish before even _starting_.

What if, instead, we fired off the RPC request for _every_ chunk at once, let them come back in whatever order they may, then reassembled them into the final array, in order.

Check out the `Promise.all` API and see if you can figure out how to implement that. You should be able to observe every RPC call start at the same time.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all